pkg/provider/okta/okta_test.go

```diff
@@ -355,3 +356,191 @@ func TestGetStateToken(t *testing.T) {
+<input type="hidden" name="eh_service_url" value=https&#x3a;&#x2f;&#x2f;1.endpointhealth.duosecurity.com&#x2f;v1&#x2f;healthapp&#x2f;device&#x2f;health&#x3f;_req_trace_group&#x3d;fa4659be389f1c724121f27a_587a98dc11576a7ab8416a32>
+<input type="hidden" name="eh_download_link" value=https&#x3a;&#x2f;&#x2f;dl.duosecurity.com&#x2f;DuoDeviceHealth-latest.pkg>
+
+<input type="hidden" name="_xsrf" value="630ef22d00af14086a3a39d62374ea80" />
+<input type="hidden" name="has_chromium_http_feature" value="true" />
+
+
+</form>
+`
+
+type testServer struct {
+	handlers []http.HandlerFunc
+	t        *testing.T
+}
+
+func (s *testServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var h http.HandlerFunc
+	if len(s.handlers) == 0 {
+		s.t.Errorf("unexpected request: %v", r.URL.String())
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	h, s.handlers = s.handlers[0], s.handlers[1:]
+	h(w, r)
+}
+func TestVerifyTrustedCert(t *testing.T) {
+	host := ""
+
+	handlers := []http.HandlerFunc{
+		func(w http.ResponseWriter, r *http.Request) {
+			assert.Equal(t, "/certifier-url", r.URL.Path)
+			assert.Equal(t, fmt.Sprintf("https://%s/", host), r.Header.Get("Referer"))
+
+			certURLRaw := r.URL.Query().Get("certUrl")
+			assert.NotEmpty(t, certURLRaw)
+			certURL, err := url.Parse(certURLRaw)
+			assert.NoError(t, err)
+
+			assert.Equal(t, "AJAX", certURL.Query().Get("type"))
+			assert.Equal(t, "HAlorymR7ZuV2CxZz9T10jABE4jzkWFBJYLEnlF4nUwCqQ=|1683218343|2a46eb852b3ef9f662304cee03d008daed6d71f6", certURL.Query().Get("sid"))
+
+			assert.Equal(t, "0dcfcbe4-5e20-47a3-9037-cb1d1bf4ad5b", certURL.Query().Get("certs_txid"))
+
+			_, err = w.Write([]byte(`{"stat":"OK"}`))
+			assert.NoError(t, err)
+		},
+
+		func(w http.ResponseWriter, r *http.Request) {
+			assert.Equal(t, "/submit", r.URL.Path)
+			assert.Equal(t, http.MethodPost, r.Method)
+			assert.Equal(t, "1234567890", r.URL.Query().Get("tx"))
+			assert.Equal(t, "2.8", r.URL.Query().Get("v"))
+
+			assert.NoError(t, r.ParseForm())
+
+			assert.Equal(t, "HAlorymR7ZuV2CxZz9T10jABE4jzkWFBJYLEnlF4nUwCqQ=|1683218343|2a46eb852b3ef9f662304cee03d008daed6d71f6", r.Form.Get("sid"))
+			assert.Equal(t, "0dcfcbe4-5e20-47a3-9037-cb1d1bf4ad5b", r.Form.Get("certs_txid"))
+			assert.Equal(t, fmt.Sprintf("https://%s/certs-url", host), r.Form.Get("certs_url"))
+			assert.Equal(t, fmt.Sprintf("https://%s/certifier-url", host), r.Form.Get("certifier_url"))
+
+		},
+	}
+
+	mockServer := &testServer{handlers: handlers, t: t}
+
+	ts := httptest.NewTLSServer(mockServer)
+	defer ts.Close()
+
+	oc, _ := setupTestClient(t, ts, "PUSH")
+	upstreamURL, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatalf("couldn't parse URL: %v", err)
+	}
+
+	host = upstreamURL.Host
+	submitURL := ts.URL + "/submit"
+
+	q := url.Values{}
+	q.Add("tx", "1234567890")
+	q.Add("parent", fmt.Sprintf("https://%s/signin/verify/duo/web", host))
+	q.Add("v", "2.8")
+
+	fakeForm := fmt.Sprintf(`
```

Comment:
originally this was a constant, but it needs the http server address and it's only available at runtime